src/cryptonote_protocol/cryptonote_protocol_defs.h

```diff
@@ -309,6 +309,10 @@ namespace cryptonote
         else
           KV_SERIALIZE_OPT(cumulative_difficulty_top64, (uint64_t)0)
         KV_SERIALIZE_CONTAINER_POD_AS_BLOB(m_block_ids)
+        // backward incompatible portable encoding
+        KV_SERIALIZE_N(m_block_weights, "m_block_weights_")
```

Comment:
Should this line be placed after line 316? The de-serialization code should try each field in-order listed here, so the value of `m_block_weights` will take precedence over (replace) `m_block_weights_`.

Reply:
It's intentional in this order, otherwise this code would be required due to how portable storage works internally:

```
         {
          if (is_store)
            KV_SERIALIZE_CONTAINER_POD_AS_BLOB(m_block_weights)
          else {
            std::vector<std::uint64_t> m_block_weights;
            epee::serialization::selector<is_store>::serialize_stl_container_pod_val_as_blob(
              m_block_weights, stg, hparent_section, "m_block_weights");
            if (this_ref.m_block_weights.empty() and not m_block_weights.empty())
              this_ref.m_block_weights = std::move(m_block_weights);
          }
        }
        KV_SERIALIZE_N(m_block_weights, "m_block_weights_")
```